src/services/votes.ts

```diff
@@ -78,13 +78,40 @@ export function numOfVotesDictionary(voteArray: Array<{ userId: string; numOfVot
   return numOfVotesDictionary;
 }
 
+export async function queryGroupCategories(
+  dbPool: PostgresJsDatabase<typeof db>,
+  questionId: string,
+): Promise<string[]> {
+  const groupCategories = await dbPool
+    .select({
+      groupCategoryId: db.questionsToGroupCategories.groupCategoryId,
+    })
+    .from(db.questionsToGroupCategories)
+    .where(eq(db.questionsToGroupCategories.questionId, questionId));
+
+  // Need to due this adjustment because currently group_category_id is nullable due to affiliations having no label.
+  const groupCategoryIds = groupCategories
+    .map((category) => category.groupCategoryId)
+    .filter((id) => id !== null) as string[];
+
+  // Returning dummy uuid if there are no group categories found
+  if (groupCategoryIds.length === 0) {
+    return ['00000000-0000-0000-0000-000000000000'];
+  }
```

Comment:
this is so sad we have to do this, lets prioritize removing this as a feature

Reply:
I agree. Assigning a group category id to affiliation in `groups` and `usersToGroups` abd treating group category as required will do the job.